File: src/集训/week3-2/F.cpp
#include <bits/stdc++.h>

using namespace std;

const int maxn = 1000000 + 100;
bool flag[maxn];
int main() {
    unsigned l, r;
    cin >> l >> r;

    unsigned t = sqrt(r);

    for (unsigned i = 2; i <= t; i++) {
        unsigned start = max(2u, (l - 1) / i + 1) * i;
        for (unsigned j = start; j <= r; j += i)
            flag[j - l] = true;
    }

    int ans = 0;
    for(int i = l; i <= r; i++)
        if(!flag[i - l])
            ans++;
    cout << ans << endl;

    return 0;
}

File: src/集训/week3-2/F.py
from math import sqrt

l, r = map(int, input().split())
n = r - l + 1
flag = [False] * n
t = int(sqrt(r))


for i in range(2, t + 1):
    start = max(2, (l - 1) // i + 1) * i
    for j in range(start, r + 1, i):
        flag[j - l] = True
cnt = 0
for isprime in flag:
    cnt += not isprime

print(cnt)